### PR TITLE
Remove redundant patterns compilations

### PR DESCRIPTION
**Pull request by gwenneg**
While looking for the root cause of a bug involving regex, I noticed a few redundant patterns compilations during `String` splits. All modified splits in this PR are called from an enclosing loop.

### CODE REVIEW

**Comment by gwenneg** [comment]
@cescoffier I also noticed there's a lot of duplicated code between `VertxCoreRecorder` and `VertxOptionsBuilder`, and I can't find any call to `VertxOptionsBuilder.buildVertxOptions()` in Quarkus. I was about to change that and then I saw `VertxOptionsBuilder` was added recently. Is there something in progress there or can I clean the duplicated code?

File: extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientRecorder.java [hunk 1]
```diff
@@ -43,6 +44,7 @@
 public class MongoClientRecorder {
 
     private static final Logger LOGGER = Logger.getLogger(MongoClientRecorder.class);
+    private static Pattern COLON_PATTERN = Pattern.compile(":");
```
Comment (gsmet) [inline extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientRecorder.java]: `final`?

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxUtil.java [hunk 2]
```diff
@@ -23,6 +24,9 @@
  * @version $Revision: 1 $
  */
 public class VertxUtil {
+
+    private static Pattern COMMA_PATTERN = Pattern.compile(",");
```
Comment (gsmet) [inline extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxUtil.java]: `final`?

File: extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java [hunk 3]
```diff
@@ -39,6 +39,8 @@
 @Recorder
 public class VertxCoreRecorder {
 
+    private static Pattern COMMA_PATTERN = Pattern.compile(",");
```
Comment (gsmet) [inline extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java]: `final`?

File: extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxOptionsBuilder.java [hunk 4]
```diff
@@ -22,6 +22,8 @@
 
 public class VertxOptionsBuilder {
 
+    private static Pattern COMMA_PATTERN = Pattern.compile(",");
```
Comment (gsmet) [inline extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxOptionsBuilder.java]: `final`?

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks good thanks, except some ``final``s are missing. Could you take care of it?

**Comment by gwenneg** [comment]
Oops, that's fixed! Thanks @gsmet.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I let @cescoffier give you an answer to your question. I think it's probably something we missed when we reorganized the Vert.x extensions.

**Comment by cescoffier** [comment]
@gwenneg yes we need to do some cleanup. The split has introduced some duplications.

**Pull request merged**